### PR TITLE
Supporting SMART App Launch with Keycloak and the IBM FHIR Server

### PR DESCRIPTION
**Pull request by lmsurpre**
my first draft for an upcoming blog post

### CODE REVIEW

File: src/pages/blog/smart-keycloak/index.mdx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+- *permission* is `read` or `write`, or `*` for read and write.
+
+SMART *launch context* is what allows a user to launch an application in the context of a given user session or patient record.
+SMART App Launch defines two different launch varieties: one for launching apps from an Electronic Health Record (EHR) system and one for launching "standalone" apps.
+
+In this post, I walk through how to support standalone app launch with the IBM FHIR Server via [Keycloak](https://www.keycloak.org), 
+an open source identity and access management solution.
+
+### Introducing the Alvearie Keycloak Extensions for FHIR
+Keycloak is wonderfully-extensible and easily-configurable.
```
Comment (mannyveiga) [inline src/pages/blog/smart-keycloak/index.mdx:31]: remove the - marks here

should be "wonderfully extensible and easily configurable"

File: src/pages/blog/smart-keycloak/index.mdx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        }
+      }
+    },
+...
+```
+
+### Conclusion 
+In this post, we walked through the OAuth 2.0 extensions defined by SMART App Launch, how to support those extensions via the Alvearie Keycloak Extensions for FHIR, and how to use this project together with the IBM FHIR Server to support SMART App Launch.
+
+SMART App Launch is just one of the specification defined by SMART-on-FHIR and the authentication and authorization challenges in healthcare are immense. We'd love to hear from members of the FHIR community, the SMART-on-FHIR community, and the Keycloak community on the approach outlined here and where you'd like to see us go next. Please connect with us on [chat.fhir.org](https://chat.fhir.org/#narrow/stream/179170-smart/topic/Keycloak.20for.20SMART.20authz), the [Keycloak Extensions for FHIR project](https://github.com/Alvearie/keycloak-extensions-for-fhir), or the [IBM FHIR Server project](https://github.com/ibm/fhir) about how we can take this to the next level.
```
Comment (mannyveiga) [inline src/pages/blog/smart-keycloak/index.mdx:167]: change to plural form - "specifications"

File: src/pages/blog/smart-keycloak/index.mdx (hunk 1 above)
Comment (mannyveiga) [inline src/pages/blog/smart-keycloak/index.mdx:31]:
```suggestion
Keycloak is wonderfully extensible and easily configurable.
```

File: src/pages/blog/smart-keycloak/index.mdx (hunk 2 above)
Comment (mannyveiga) [inline src/pages/blog/smart-keycloak/index.mdx:167]:
```suggestion
SMART App Launch is just one of the specifications defined by SMART-on-FHIR and the authentication and authorization challenges in healthcare are immense. We'd love to hear from members of the FHIR community, the SMART-on-FHIR community, and the Keycloak community on the approach outlined here and where you'd like to see us go next. Please connect with us on [chat.fhir.org](https://chat.fhir.org/#narrow/stream/179170-smart/topic/Keycloak.20for.20SMART.20authz), the [Keycloak Extensions for FHIR project](https://github.com/Alvearie/keycloak-extensions-for-fhir), or the [IBM FHIR Server project](https://github.com/ibm/fhir) about how we can take this to the next level.
```

**Review by mannyveiga — CHANGES_REQUESTED** [review]

File: src/pages/blog/smart-keycloak/index.mdx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+...
+```
+
+### Conclusion 
+In this post, we walked through the OAuth 2.0 extensions defined by SMART App Launch, how to support those extensions via the Alvearie Keycloak Extensions for FHIR, and how to use this project together with the IBM FHIR Server to support SMART App Launch.
+
+SMART App Launch is just one of the specifications defined by SMART-on-FHIR and the authentication and authorization challenges in healthcare are immense. We'd love to hear from members of the FHIR community, the SMART-on-FHIR community, and the Keycloak community on the approach outlined here and where you'd like to see us go next. Please connect with us on [chat.fhir.org](https://chat.fhir.org/#narrow/stream/179170-smart/topic/Keycloak.20for.20SMART.20authz), the [Keycloak Extensions for FHIR project](https://github.com/Alvearie/keycloak-extensions-for-fhir), or the [IBM FHIR Server project](https://github.com/ibm/fhir) about how we can take this to the next level.
+
+---
+SMART is the registered trademarks of SMART Technologies ULC in the U.S. and/or other countries.
```
Comment (prb112) [inline src/pages/blog/smart-keycloak/index.mdx:170]: Perhaps it should follow https://smarthealthit.org/smart-logos-guidelines-for-trademark-use/
I don't think the above is accurate.

Comment (lmsurpre) [reply src/pages/blog/smart-keycloak/index.mdx:170]: Good catch.  I just googled SMART trademark and followed https://www.smarttech.com/en/legal/trademark-guidelines...didn't even realize that there are two entirely different groups claiming trademark on "SMART"...seems problematic.

Comment (lmsurpre) [reply src/pages/blog/smart-keycloak/index.mdx:170]:
```suggestion
SMART is a trademark of The Children’s Medical Center Corporation.
```

File: src/pages/blog/smart-keycloak/index.mdx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+### Using the Alvearie Keycloak Extensions for FHIR
+The Keycloak extensions for FHIR project currently consists of two components:
+- *keycloak-config* for automating the configuration of a Keycloak realm in support of standalone app launch with patient context
+- *keycloak-extensions* which defines a set of extensions that fill the gaps in Keycloak's support of SMART App Launch:
+  - an AudienceValidator authenticator for validating that the `aud` parameter passed to the server matches the desired value
+  - a UserAttributeMapper mapper for mapping user attributes into custom fields in the token response payload (rather than claims in the issued tokens)
+
+To use the extensions:
+1. Build the keycloak-extensions project into a jar file and add it to the [`standalone/deployments`](https://www.keycloak.org/docs/latest/server_development/#packaging-classes-and-deployment-2) directory of your Keycloak installation.
+    - Note: moving forward, we hope to add continuous integration support and publish the built artifacts to Maven Central.
```
Comment (lmsurpre) [inline src/pages/blog/smart-keycloak/index.mdx:50]:
```suggestion
    - Note: moving forward, we hope to publish the built artifacts to Maven Central.
```

**Pull request merged**